chore: refactor of styling logic (#128)
spryker/oryx: docs/drafts/styling/typography.md

**note**: While `rem` (_root_ `em`) is a fair alternative to `em` it does not work out of the box in oryx, as oryx does not control the _root_ of the page. `rem`-based sizes can however be configured and controlled in themes.

### Semantic HTML structure versus UI

The semantic structure of the typography elements is important on the web. Crawlers and screen readers use the structure to interpreted the content. This is important for users with limited sight, as they will be guided by screen readers. It helps navigating the content and allows to skip sections which are not of interest.

If the structure contains an incorrect order, e.g. a H3 followed by a H5, it violates with accessibility best practices.

This strict structure might not be required for the UI design, as the layout of the page often helps with structuring content. UI designers often ignore the structure of the page, and use heading levels in a random order, just to satisfy their design needs. This might be fair, as visual layout accommodates the structure, so from a pure design perspective, there's often a different perspective on the heading levels.

> To allow for a flexible but compliant design, the heading component supports to distinguish the structure from the UI when needed. The heading component is equipped with an utility to mimic the structure _as_ with alternative typography.

Reviewer: > The heading component is equipped with an utility to mimic the structure _as_ with alternative typography.

It's clear what `as` from examples below, but I'd say it's a bit difficult to understand it from this sentence.

```html
<oryx-heading as="h3">

>   <h1>This h1 appears as an h3 on small screens</h1>

Reviewer: ```suggestion
  <h1>This heading appears as an h3</h1>
```

</oryx-heading>
```

The mimicking can be applied per screen size (lg, md, sm), for example if you only want to change the UI on small screens you could use:

```html
<oryx-heading as-sm="h3">

>   <h1>This h1 appears as an h3 on small screens</h1>

Reviewer: ```suggestion
  <h1>This heading appears as an h3 on small screens</h1>
```

The last option is to provide the heading appearance, regardless of the actual `tag`. This will render the tag as-is, but applies the styles based on the UI design.

```html
<oryx-heading as="h5">

>   <h3>This heading appears as an H5</h3>

Reviewer: ```suggestion
  <h3>This heading appears as an h5</h3>
```

</oryx-heading>

> <oryx-heading as="h5"> This text appears as an H5 </oryx-heading>

Reviewer: ```suggestion
<oryx-heading as="h5"> This text appears as an h5 </oryx-heading>
```

Reviewer: I'd suggest to check it everywhere so to have lowercased version.

<oryx-heading tag="h3" as="h5">

>   This text renders as an h3, but appears as an H5

Reviewer: ```suggestion
  This text renders as an h3, but appears as an h5
```
